File: frontend/src/services/notificationService.js
import api from "./api";

export const notificationService = {

  // Notifications

  getNotifications: () =>
    api
      .get("/notifications")
      .then((response) => response.data),

  getUnreadCount: () =>
    api
      .get("/notifications/unread-count")
      .then((response) => response.data.count),

  markAsRead: (id) =>
    api
      .put(`/notifications/${id}/read`)
      .then((response) => response.data),

  markAllAsRead: () =>
    api
      .put("/notifications/read-all")
      .then((response) => response.data),

  deleteNotification: (id) =>
    api
      .delete(`/notifications/${id}`)
      .then((response) => response.data),

  // Reminder Settings

  getSettings: () =>
    api
      .get("/notifications/settings")
      .then((response) => response.data),

  updateSettings: (payload) =>
    api
      .put("/notifications/settings", payload)
      .then((response) => response.data),

};

File: backend/src/main/java/com/paisatrack/backend/service/NotificationService.java
package com.paisatrack.backend.service;

import com.paisatrack.backend.dto.NotificationResponse;
import com.paisatrack.backend.dto.NotificationSettingRequest;
import com.paisatrack.backend.dto.NotificationSettingResponse;

import java.util.List;

public interface NotificationService {

        // ==========================================
        // Notifications
        // ==========================================

        List<NotificationResponse> getNotifications();

        Long getUnreadCount();

        void markAsRead(Long notificationId);

        void markAllAsRead();

        void deleteNotification(Long notificationId);

        // ==========================================
        // Reminder Settings
        // ==========================================

        NotificationSettingResponse getSettings();

        NotificationSettingResponse saveSettings(
                NotificationSettingRequest request
        );

        // ==========================================
        // Scheduler Support
        // ==========================================

        void createReminderNotification(
                Long userId,
                String title,
                String message,
                String type
        );

        void createSystemNotification(
                Long userId,
                String title,
                String message,
                String type
        );
}